### ISSUE TITLE
Missing accessible name on progressbar references WCAG SC 1.1.1 instead of 4.1.2

### ISSUE THREAD

**Issue by yatil** [dequelabs/axe-core#4282]
### Product

axe-core

### Latest Version

- [X] I have tested the issue with the latest version of the product

#### Expectation

When a progressbar element (`role="progressbar"`) has no accessible name, I expect this to be a [SC 4.1.2 Name, Role, Value](https://www.w3.org/TR/WCAG21/#name-role-value) violation

#### Actual

The violation is categorized as a [SC 1.1.1 Non-text Content](https://www.w3.org/TR/WCAG21/#non-text-content) violation. 

#### How to Reproduce

Create a page with an aria-progress bar, do not label it. Run axe-core.

#### Additional context

See the Pull Request where the test was added:

- https://github.com/dequelabs/axe-core/pull/2555

**Comment by WilcoFiers** [comment]
Thank you for opening this issue. I don't think there's a definitive answer to this. The way I read 4.1.2, it applies to user interface components, which is defined as something perceived as a single control by users. When I think of a control I think of something I can interact with. Something that lets me do something. Progress bars provide information, but they aren't (usually) interactive. Since they pretty much by their nature have to provide information without text, I think it fits better under "non-text content" than "user interface component".

I get where you're coming from. 4.1.2 is the SC where most of the ARIA stuff goes. But this was an intentional decision.

**Comment by yatil** [comment]
The superclass role of a [progressbar](https://w3c.github.io/aria/#progressbar) is a [widget](https://w3c.github.io/aria/#widget) which is, quote:

> An interactive component of a graphical user interface (GUI).

As the value can change, this IMHO qualifies as a user interface component. At least both, 4.1.2 and 1.1.1 should be cited, even if 1.1.1 is redundant.

**Comment by WilcoFiers** [comment]
(Sorry vacation). In what way are progress bars interactive?

**Comment by yatil** [comment]
That’s the definition in WAI-ARIA. I linked to the relevant document. I cannot name the reasons for the ARIA WG’s decisions.

A progress bar often changes based on user input or a user action, even if not directly with that element. I think that’s interactive enough for me. That said, I don’t think I care enough.
